src/System.Private.CoreLib/shared/System/Buffers/StandardFormat.cs

```diff
@@ -146,40 +146,47 @@ private static bool ParseHelper(ReadOnlySpan<char> format, out StandardFormat st
         /// </summary>
         public override string ToString()
         {
-            unsafe
+            Span<char> buffer = stackalloc char[FormatStringLength];
+            int charsWritten = Format(buffer);
+            return new string(buffer.Slice(0, charsWritten));
+        }
+
+        /// <summary>The length buffer needed by <see cref="Format"/>.</summary>
+        internal const int FormatStringLength = 4;
```

Comment:
This could be 3, for example G99, or do we want to keep it a power of 2?

Reply:
Yes, once we change Format to not have that >= 100 case, this drops to 3.